HousingRepairsOnlineApi/UseCases/RetrieveRepairsUseCase.cs

```diff
@@ -25,5 +27,22 @@ public async Task<IEnumerable<RepairRequestSummary>> Execute(string repairType,
 
             return await cosmosGateway.SearchByPropertyReference(repairType, propertyReference);
         }
+
+        public async Task<Repair> Execute(IEnumerable<string> repairTypes, string postcode, string repairId)
+        {
+            Guard.Against.NullOrEmpty(repairTypes, nameof(repairTypes));
+            Guard.Against.NullOrWhiteSpace(postcode, nameof(postcode));
+            Guard.Against.NullOrWhiteSpace(repairId, nameof(repairId));
+
+            var searchTask = cosmosGateway.SearchByPostcodeAndId(repairTypes, postcode, repairId);
+
+            var result = await searchTask.ContinueWith(x =>
+            {
+                var repairRequestSummaries = x.Result.ToArray();
+                return repairRequestSummaries.Length == 1 ? repairRequestSummaries[0] : null;
```

Comment:
What if there are 2 entries returned? Shouldn't we return one of them rather than saying that the repair cannot be found?

Reply:
If there are multiple entries found, it would be better to return nothing to _potentially_ incorrect information is not shown, i.e. if there are multiple matches with differing details (appointment times, location, problem, etc), the system does not know how to determine which is correct and therefore should not return anything.

The occurrence of multiple items being found is highly unlikely.